packages/reselect/src/index.d.ts

```diff
@@ -0,0 +1,1108 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
```

Comment:
Do we need this file if we're leaving the `reselect` library overrides in JS?

Reply:
It's presumably needed to stop TS from complaining about us calling `createSelector` incorrectly since we added an extra name parameter to its result

Reply:
Yes, typescript was complaining if I didn't have it unfortunately